R/loadFromMetaphlan.R

```diff
@@ -0,0 +1,226 @@
+#' Import Metaphlan results to \code{TreeSummarizedExperiment}
+#
+#' @param file a single \code{character} value defining the file
+#'   path of the Metaphlan file. The file must be in merged Metaphlan format.
+#'
+#' @param sample_meta a single \code{character} value defining the file
+#'   path of the sample metadata file. The file must be in \code{tsv} format
+#'   (default: \code{sample_meta = NULL}).
+#'   
+#' @param phy_tree a single \code{character} value defining the file
+#'   path of the phylogenetic tree.
+#'   (default: \code{phy_tree = NULL}).
+#'   
+#' @param ... additional arguments:
+#' \itemize{
+#'   \item{\code{abund_values}:} {A single character value for naming 
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}} 
+#'   (default: \code{abund_values = "counts})}
+#'   \item{\code{removeTaxaPrefixes}:} {\code{TRUE} or \code{FALSE}: Should
+#'     taxonomic prefixes be removed? (default:
+#'     \code{removeTaxaPrefixes = FALSE})}
+#' }
+#'
+#' @details
+#' Import Metaphlan results. Input must be in merged Metaphlan format.
+#' Data is imported so that data at the lowest rank is imported as a 
+#' \code{TreeSE} object. Data at higher rank is imported as a
+#' \code{SE} objects which are stored to \code{altExp} of
+#' \code{TreeSE} object.
+#'
+#' @return  A
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' object
+#'
+#' @name loadFromMetaphlan
+#' @seealso
+#' \code{\link[=makeTreeSummarizedExperimentFromPhyloseq]{makeTreeSummarizedExperimentFromPhyloseq}}
+#' \code{\link[=makeSummarizedExperimentFromBiom]{makeSummarizedExperimentFromBiom}}
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{makeTreeSummarizedExperimentFromDADA2}}
+#' \code{\link[=loadFromQIIME2]{loadFromQIIME2}}
+#' \code{\link[=loadFromMothur]{loadFromMothur}}
+#'
+#' @export
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#' 
+#' @references
+#'
+#' @examples
+#' \dontrun{
+#' # File path
+#' file_path <- "/data/metaphlan_result.txt"
```

Comment:
If the file is not large, it could be included as example data in `inst/extdata/` and called with `system.file()`. Then the example becomes reproducible and you can remove `dontrun`

Reply:
@KatariinaParnanen has promised to provide another open demo data set for this

Reply:
I added the example file. It's not too big, it's 8 KB while there are files 2-90 KB in extdata